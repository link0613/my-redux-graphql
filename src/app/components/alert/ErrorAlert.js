// @flow

// #region imports
import React, { PureComponent } from 'react';
import { Motion, spring, presets } from 'react-motion';
// #endregion

// #region flow types
export type Props = {
  showAlert: boolean,
  errorTitle: string,
  errorMessage: string,
  onClose: () => any,
};

export type State = {};
// #endregion

class ErrorAlert extends PureComponent<Props, State> {
  static defaultProps = {
    showAlert: false,
  };

  // #region lifecycle methods
  render() {
    const { showAlert, errorTitle, errorMessage, onClose } = this.props;

    return (
      <Motion
        style={{
          interpolatedScale: spring(showAlert ? 1 : 0, presets.stiff),
        }}
      >
        {({ interpolatedScale }) => (
          <div
            className="alert alert-dismissible alert-danger"
            style={{
              WebkitTransform: `scale(${interpolatedScale})`,
              transform: `scale(${interpolatedScale})`,
            }}
          >
            <button type="button" className="close" onClick={onClose}>
              &times;
            </button>
            {errorTitle &&
              errorTitle.length > 0 && <strong>{errorTitle}</strong>}
            {<p>{errorMessage}</p>}
          </div>
        )}
      </Motion>
    );
  }
  // #endregion
}

export default ErrorAlert;
