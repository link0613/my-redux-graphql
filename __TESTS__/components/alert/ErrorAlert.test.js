import React from 'react';
import ErrorAlert from '../../../src/app/components/alert/ErrorAlert';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { mount } from 'enzyme';

describe('ErrorAlert component', () => {
  it('renders as expected', () => {
    const props = { showAlert: true, type: 'warning' };

    const component = renderer
      .create(
        <ErrorAlert {...props}>
          <p>an error alert</p>
        </ErrorAlert>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('renders as an error alert', () => {
    const props = {
      showAlert: true,
      type: 'error',
      errorTitle: 'error title',
      errorMessage: 'error message',
    };

    const component = mount(
      <ErrorAlert {...props}>
        <p>an alert</p>
      </ErrorAlert>,
    );

    expect(component.find('div.alert')).toBeDefined();
    expect(component.find('div.alert').hasClass('alert-danger'));
  });

  it('should call onClose on a close button click', () => {
    const onClickMock = jest.fn();
    const preventDefault = jest.fn();

    const mockClickEvent = { preventDefault };

    const component = mount(
      <ErrorAlert showAlert={true} type="warning" onClose={onClickMock} />,
    );

    const button = component.find('button.close');
    button.simulate('click', mockClickEvent);

    expect(onClickMock).toHaveBeenCalled();
  });

  it('should be scaled to zero', () => {
    const props = {
      showAlert: false,
    };

    const component = mount(<ErrorAlert {...props} />);
    const divAlert = component.find('div.alert');
    expect(divAlert).toBeDefined();
    expect(divAlert.prop('style')).toHaveProperty('transform', 'scale(0)');
  });
});
