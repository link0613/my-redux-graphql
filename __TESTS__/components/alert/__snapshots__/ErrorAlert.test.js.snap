// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorAlert component renders as expected 1`] = `
<div
  className="alert alert-dismissible alert-danger"
  style={
    Object {
      "WebkitTransform": "scale(1)",
      "transform": "scale(1)",
    }
  }
>
  <button
    className="close"
    onClick={undefined}
    type="button"
  >
    Ã—
  </button>
  <p />
</div>
`;
